### DISCUSSION TITLE
Whenever I restart the docker container, DALL-E generated images are lost

### DISCUSSION THREAD

**Discussion by raymelon** [danny-avila/LibreChat#3537] [Troubleshooting] [answered]
### What happened?

Whenever I restart the docker container, DALL-E generated images are lost

I am using the default template in Zeabur

### Steps to Reproduce

1. Generate an image with DALL-E plugin
2. See the image included in Files (this happens automatically)
3. Restart LibreChat container
4. See the images disappear

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
rag-api output:

2024-08-04 22:00:54,656 - uvicorn.error - INFO - Started server process [1]

2024-08-04 22:00:54,657 - uvicorn.error - INFO - Waiting for application startup.

2024-08-04 22:00:54,657 - uvicorn.error - INFO - Application startup complete.

2024-08-04 22:00:54,658 - uvicorn.error - INFO - Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)

2024-08-04 22:00:57,037 - uvicorn.error - INFO - Shutting down

2024-08-04 22:00:57,138 - uvicorn.error - INFO - Waiting for application shutdown.

2024-08-04 22:00:57,139 - uvicorn.error - INFO - Application shutdown complete.

2024-08-04 22:00:57,139 - uvicorn.error - INFO - Finished server process [1]

2024-08-04 22:01:05,870 - root - INFO - Initialized embeddings of type: <class 'langchain_openai.embeddings.base.OpenAIEmbeddings'>

/usr/local/lib/python3.10/site-packages/langchain_community/vectorstores/pgvector.py:322: LangChainPendingDeprecationWarning: Please use JSONB instead of JSON for metadata. This change will allow for more efficient querying that involves filtering based on metadata.Please note that filtering operators have been changed when using JSOB metadata to be prefixed with a $ sign to avoid name collisions with columns. If you're using an existing database, you will need to create adb migration for your metadata column to be JSONB and update your queries to use the new operators. 

  warn_deprecated(

LibreChat output:

2024-08-04 22:01:28 warn: Default value for CREDS_KEY is being used.

2024-08-04 22:01:28 warn: Default value for CREDS_IV is being used.

2024-08-04 22:01:28 info: Please replace any default secret values.

2024-08-04 22:01:28 info: ​


    For your convenience, use this tool to generate your own secret values:

    https://www.librechat.ai/toolkit/creds_generator


    ​

2024-08-04 22:01:29 info: RAG API is running and reachable at http://rag-api:8000.

2024-08-04 22:01:29 info: Server listening on all interfaces at port 3080. Use http://localhost:3080 to access it
```

### Screenshots

Images are lost after I restarted the service

![image](https://github.com/user-attachments/assets/a72ecdf8-7a8c-419d-b53d-40eecf4549d9)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
You need to mount the images volume. in both of the project's provided compose files, this is done here:

```yaml
    volumes:
      - type: bind
        source: ./.env
        target: /app/.env
      - ./images:/app/client/public/images
```

**Reply by raymelon** [reply]
Hey @danny-avila! As always, I appreciate the prompt response!

I am currently exploring your suggestion.
It seems like there is an issue in Zeabur template in terms of mounting volumes.

I confirmed this by using a local build that has a `docker-compose.yml` that mounts the volumes correctly.
Local build is showing some images, meanwhile the Zeabur instance does not show any.

I am assuming that all the images that are loading on localhost build are generated in localhost.

Sending the screenshots here for reference:

localhost build:
![image-zeabur](https://github.com/user-attachments/assets/b59b6ca0-1be7-43fe-a63e-67c66fdb9a4d)

Zeabur build:
![image-zeabur-2](https://github.com/user-attachments/assets/8ca8c639-7cce-47a2-8645-8298815d2ccc)

Will report with my findings as I explore.
If there is something I'm missing, please let me know.

**Reply by danny-avila** [reply]
no that makes sense, I'm not too familiar with Zeabur. As an alternative, you could use firebase as a storage layer/CDN for images.

https://www.librechat.ai/docs/configuration/firebase

Would like to offer S3 soon

**Reply by raymelon** [reply]
Love it! Will check Firebase out! 

So I am assuming that the Files sidebar is a combination of RAG files and mounted images?

Does it mean that both uploaded images and images generated by DALL-E are not stored in RAG?

**Reply by raymelon** [reply]
Just confirmed from Zeabur Discord that `./images:/app/client/public/images` is missing in `volumes`

Zeabur just updated their template.

Appreciate the help @danny-avila!

**Reply by danny-avila** [reply]
> Does it mean that both uploaded images and images generated by DALL-E are not stored in RAG?

Correct, only images are stored as files, everything else is stored as vectors for the rag api. Technically the images could be vectorized though

> Just confirmed from Zeabur Discord that ./images:/app/client/public/images is missing in volumes

Awesome!

**Reply by raymelon** [reply]
> > Does it mean that both uploaded images and images generated by DALL-E are not stored in RAG?
> 
> Correct, only images are stored as files, everything else is stored as vectors for the rag api. Technically the images could be vectorized though
> 
> > Just confirmed from Zeabur Discord that ./images:/app/client/public/images is missing in volumes
> 
> Awesome!

Cool! Thanks for clarifying!
